lib/plugins/sfDoctrinePlugin/lib/validator/sfValidatorDoctrineChoice.class.php

```diff
@@ -76,6 +77,11 @@ protected function doClean($value)
 
       $count = count($value);
 
+      if (array_keys($value) !== range(0, $count-1))
+      {
+        throw new sfValidatorError($this, 'invalid_keys', array('keys' => array_keys($value)));
```

Comment:
This is a technical bug and not an user one. Than I propose to just use `array_values()` to have a valid array for Doctrine.

Note: We need to be sure that the array is not empty because on this case all records will be fetched. Then is a performance issue there.